spec/db/data/metabase_user_restrictions_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+describe "RestrictiveAccessForMetabaseUser" do
+  before(:all) do
+    Rails.application.load_tasks
+  end
+
+  context "when attempting intentional behavior" do
+    it "can select tables from the analytics table" do
+      RSpec::Expectations.configuration.on_potential_false_positives = :nothing
+
+      expect {
+        ActiveRecord::Base.connection.execute("
+set role metabase;
+select count(*) from analytics.team_role_members (vita_partner_id, id);
+select * from analytics.users where failed_attempts == 0;
+")
+      }.not_to raise_exception(ActiveRecord::StatementInvalid, /permission denied/)
+
+    end
+  end
+
+  context "when attempting unintentional behavior" do
+    before(:all) do
```

Comment:
I think this `before(:all)` should probably be in the outermost block of the file to ensure it re-creates the user for both the positive and negative tests